packages/core/src/types/oas2.ts

```diff
@@ -353,9 +350,11 @@ const SecurityScheme: NodeType = {
         return ['type', 'flow'];
       }
     }
-
     return ['type'];
   },
+  allowed(value) {
+    return value.type === 'basic' ? ['type'] : [];
```

Comment:
For me it reads: for type basic only `type` is allowed, for other types - nothing is allowed. Maybe change it to:

```suggestion
    return value.type === 'basic' ? ['type'] : undefined
```